go/spec/config_test.go

```diff
@@ -0,0 +1,182 @@
+package spec
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/attic-labs/noms/go/d"
+	"github.com/attic-labs/testify/assert"
+)
+
+const (
+	ldbSpec    = "ldb:./local"
+	memSpec    = "mem"
+	httpSpec   = "http://test.com:8080/foo"
+	remoteAlias = "origin"
+)
+
+var (
+	ctestRoot = os.TempDir()
+
+	ldbConfig = &Config{
+		"",
+		DefaultConfig{ ldbSpec },
+		map[string]DbConfig{ remoteAlias: { httpSpec }},
+	}
+
+	httpConfig = &Config{
+		"",
+		DefaultConfig{ httpSpec },
+		map[string]DbConfig{ remoteAlias: { ldbSpec }},
+	}
+
+	memConfig = &Config{
+		"",
+		DefaultConfig{ memSpec },
+		map[string]DbConfig{ remoteAlias: { httpSpec }},
+	}
+)
+
+type paths struct {
+	home string
+	config string
+}
+
+func getPaths(base string) paths {
+	abs, err := filepath.Abs(ctestRoot)
+	d.PanicIfError(err)
```

Comment:
I know our error handling is kind of messed up, but I'm not sure it's worth it to use PanicIfError in test cases since that wraps the error with stuff that might not be that useful in test case failure reporting.